### PR TITLE
fix: issue #2735 unused axis argument in from_ndarray generator

### PR DESCRIPTION
**Pull request by valemore**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200479679796904/1200533970296212) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by maximilianwerk** [comment]
Dear @valemore,

thanks a lot for your contribution. could you add a test in order to showcase, what you are solving? Thanks a lot :)

**Comment by valemore** [comment]
> Dear @valemore,
> 
> thanks a lot for your contribution. could you add a test in order to showcase, what you are solving? Thanks a lot :)

Hi @maximilianwerk, I added a unit test that previously would have failed because the axis argument to from_ndarray has no effect.

File: tests/unit/test_generators.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import numpy as np
+
+from jina.types.document.generators import from_ndarray
+
+def test_from_ndarray():
+    test_array = np.random.random((10, 64))
+    # yield using from_ndarray along axis 0
+    test_generator = from_ndarray(test_array)
+    assert next(iter(test_generator)).blob.shape == np.array([64])
+
```
Comment (JoanFM) [inline tests/unit/test_generators.py:10]: why is iter needed?

File: tests/unit/test_generators.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # yield using from_ndarray along axis 1
+    test_array = test_array.T
+    test_generator = from_ndarray(test_array, axis=1)
```
Comment (JoanFM) [inline tests/unit/test_generators.py:13]: shouldnt here the shape be 10?

Comment (JoanFM) [reply tests/unit/test_generators.py:13]: I see now, the transpose. May u parametrize better the input and the expected shape? this way the test would be more readable?

File: tests/unit/test_generators.py (hunk 1 above)
Comment (valemore) [reply tests/unit/test_generators.py:10]: Not needed indeed, it's redundant :)

File: tests/unit/test_generators.py (hunk 2 above)
Comment (valemore) [reply tests/unit/test_generators.py:12]: yes, the transpose can be missed. Parametrized like this it's maybe more explicit.

**Review by valemore — COMMENTED** [review]
updated as per @JoanFM 's feedback

**Comment by nan-wang** [comment]
@valemore I'm afraid you need to rebase your codes to pass the CI. https://github.com/jina-ai/jina/pull/2788/checks?check_run_id=2988606249#step:4:9

```
git checkout master
git pull 
git checkout fix-generators-2735
git reset --soft <last_commit_your_branch_share_with_master>
git commit -m 'fix: fix the issue 2735'
git rebase master
git push -f origin fix-targepeapod-2910
```

File: jina/types/document/generators.py [hunk 3]
```diff
@@ -31,9 +31,9 @@ def from_ndarray(
 
     if shuffle:
         # shuffle for random query
-        array = np.take(array, np.random.permutation(array.shape[0]), axis=axis)
+        array = np.take(array, np.random.permutation(array.shape[axis]), axis=axis)
```
Comment (bwanglzu) [inline jina/types/document/generators.py:34]: maybe it's even better to:
```suggestion
        array = np.random.shuffle(array) 
```

and parameterize the unit test

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: jina/types/document/generators.py (hunk 3 above)
Comment (bwanglzu) [reply jina/types/document/generators.py:34]: and remove the `axis`, i don't see why it's needed

**Comment by bwanglzu** [comment]
AFA2C3B2-4FA0-47EC-BDAF-6292B094E2C2
if we decided to go with `np.take`, please ignore my comments above, but i found `shuffle` more efficient.

**Pull request closed without merging**